pkg/sources/forward/data_forward.go

```diff
@@ -192,6 +195,17 @@ type readWriteMessagePair struct {
 	transformerError error
 }
 
+// emitMessageSizeMetric emits the message size metric
+func emitMessageSizeMetric(df *DataForward, m *isb.ReadMessage) {
```

Comment:
I don't think we need to create a separate function for `ReadBytesCount`. For 2 reasons: 1. it's inconsistent with how we emit other metrics, one can argue if we create func for ReadBytesCount, why not all other metrics? 2. and mainly, functions should be [deep](https://medium.com/@alexturdean/shallow-functions-the-silent-killer-of-good-software-527a29ab56c9).

Reply:
+1